src/rust/receiver-mock/src/router.rs

```diff
@@ -437,8 +437,7 @@ pub async fn handler_receiver(
 
     // parse the value of the X-Sumo-Fields header
     // TODO: use the metadata for metrics
-    let x_sumo_fields_value = headers.get("x-sumo-fields").unwrap_or(&empty_header).to_str().unwrap();
-    let metadata = match parse_sumo_fields_header_value(x_sumo_fields_value) {
+    let metadata = match get_metadata_from_headers(headers) {
```

Comment:
This (fields from headers - `X-Sumo-Fields`) works only for logs. Let's do it only in the log related branch in here then

Reply:
Hm, for `X-Sumo-Fields`, yes, but don't metrics and traces still want the other fields?